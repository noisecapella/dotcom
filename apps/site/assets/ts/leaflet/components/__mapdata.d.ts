import { ReactElement } from "react";
import { CrowdingType } from "../../schedule/components/__schedule";

export type TileServerUrl =
  | "https://cdn.mbta.com"
  | "https://mbta-map-tiles-dev.s3.amazonaws.com"
  | "";

export interface IconOpts {
  icon_size?: [number, number];
  icon_anchor?: [number, number];
  popup_anchor?: [number, number];
}

export interface MapMarker {
  icon: string | null;
  icon_opts?: IconOpts;
  id: string | null;
  longitude: number;
  latitude: number;
  size?: number[];
  rotation_angle: number;
  tooltip: ReactElement<HTMLElement> | null;
  tooltip_text?: string | null;
  z_index?: number;
  shape_id?: string;
  vehicle_crowding?: CrowdingType;
  stop_id?: string;
  vehicle_status?: "in_transit" | "incoming" | "stopped";
}

export interface Polyline {
  color: string;
  "dotted?": boolean;
  id: string;
  positions: [number, number][];
  weight: number;
}

export interface MapData {
  id?: string;
  default_center: {
    longitude: number;
    latitude: number;
  };
  height: number;
  markers: MapMarker[];
  stop_markers?: MapMarker[];
  polylines: Polyline[];
  tile_server_url: TileServerUrl;
  width: number;
  zoom: number | null;
}

export interface StaticMapData {
  img_src: string;
  pdf_url: string;
}
