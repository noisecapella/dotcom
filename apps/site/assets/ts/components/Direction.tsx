import React, { ReactElement } from "react";
import Headsigns from "./Headsign";
import { Direction as DirectionType, EnhancedRoute } from "../__v3api";

interface Props {
  direction: DirectionType;
  route: EnhancedRoute;
}

export const directionIsEmpty = (dir: DirectionType): boolean =>
  dir.headsigns.length === 0;

export const Direction = ({
  direction,
  route
}: Props): ReactElement<HTMLElement> | null => {
  if (directionIsEmpty(direction)) {
    return null;
  }

  const condensed = direction.headsigns.length === 1;

  const hideDirectionDestination =
    condensed || route.type === 0 || route.type === 1 || route.type === 2;

  return (
    <div className="m-tnm-sidebar__direction">
      <div className="m-tnm-sidebar__direction-name u-small-caps">
        {route.direction_names[direction.direction_id]}
      </div>
      {!hideDirectionDestination && (
        <div className="m-tnm-sidebar__direction-destination">
          {route.direction_destinations[direction.direction_id]}
        </div>
      )}
      <div className="m-tnm-sidebar__direction-headsigns">
        <Headsigns
            key={`headsigns-${route.direction_destinations[direction.direction_id]}`}
            routeType={route.type}
            condensed={condensed}
            headsigns={direction.headsigns}
          />
        {/* {direction.headsigns.map(headsign => (
          <Headsign
            key={headsign.headsign_name!}
            routeType={route.type}
            condensed={condensed}
            {...headsign}
          />
        ))} */}
      </div>
    </div>
  );
};

export default Direction;
