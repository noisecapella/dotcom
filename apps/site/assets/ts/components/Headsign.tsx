import React, { ReactElement } from "react";
import { RouteType, PredictedOrScheduledTime, Headsign } from "../__v3api";
import {
  timeForCommuterRail,
  statusForCommuterRail,
  trackForCommuterRail
} from "../helpers/prediction-helpers";

interface Props {
  headsign: Headsign;
  routeType: RouteType;
  condensed: boolean;
}

const headsignClass = (condensed: boolean): string => {
  if (condensed === true) {
    return "m-tnm-sidebar__headsign-schedule m-tnm-sidebar__headsign-schedule--condensed";
  }
  return "m-tnm-sidebar__headsign-schedule";
};

const renderHeadsignName = ({
  headsign,
  routeType,
  condensed
}: Props): ReactElement<HTMLElement> => {
  const modifier = !condensed && routeType === 3 ? "small" : "large";

  const headsignNameClass = `m-tnm-sidebar__headsign-name m-tnm-sidebar__headsign-name--${modifier}`;

  const headsignName = headsign.headsign || headsign.name;
  if (headsignName && headsignName.includes(" via ")) {
    const split = headsignName.split(" via ");
    return (
      <>
        <div className={headsignNameClass}>{split[0]}</div>
        <div className="m-tnm-sidebar__via">{`via ${split[1]}`}</div>
      </>
    );
  }
  return <div className={headsignNameClass}>{headsignName}</div>;
};

const renderTrainName = (trainName: string): ReactElement<HTMLElement> => (
  <div className="m-tnm-sidebar__headsign-train">{trainName}</div>
);

const renderTimeCommuterRail = (
  data: PredictedOrScheduledTime,
  modifier: string
): ReactElement<HTMLElement> => {
  const status = statusForCommuterRail(data) || "";
  return (
    <div
      className={`m-tnm-sidebar__time m-tnm-sidebar__time--commuter-rail ${modifier} ${
        status === "Scheduled" ? "text-muted" : ""
      }`}
    >
      {timeForCommuterRail(
        data,
        `${
          status === "Canceled" ? "strikethrough" : ""
        } m-tnm-sidebar__time-number`
      )}
      <div className="m-tnm-sidebar__status">
        {`${status}${trackForCommuterRail(data)}`}
      </div>
    </div>
  );
};

const renderTimeDefault = (
  time: string[],
  modifier: string
): ReactElement<HTMLElement> => (
  <div className={`m-tnm-sidebar__time ${modifier}`}>
    <div className="m-tnm-sidebar__time-number">{time[0]}</div>
    <div className="m-tnm-sidebar__time-mins">{time[2]}</div>
  </div>
);

const renderTime = (
  tnmTime: PredictedOrScheduledTime,
  headsignName: string,
  routeType: RouteType,
  idx: number
): ReactElement<HTMLElement> => {
  // eslint-disable-next-line camelcase
  const { prediction, scheduled_time } = tnmTime;
  // eslint-disable-next-line camelcase
  const time = prediction ? prediction.time : scheduled_time!;

  const classModifier =
    !prediction && [0, 1, 3].includes(routeType)
      ? "m-tnm-sidebar__time--schedule"
      : "";

  return (
    <div
      // eslint-disable-next-line camelcase
      key={`${headsignName}-${idx}`}
      className="m-tnm-sidebar__schedule"
    >
      {routeType === 2
        ? renderTimeCommuterRail(tnmTime, classModifier)
        : renderTimeDefault(time, classModifier)}
    </div>
  );
};

const HeadsignComponent = (props: Props): ReactElement<HTMLElement> => {
  const { headsign, routeType, condensed } = props;
  return (
    <div className={headsignClass(condensed)}>
      <div className="m-tnm-sidebar__headsign">
        {renderHeadsignName(props)}

        {routeType === 2 && headsign.train_number
          ? renderTrainName(`Train ${headsign.train_number}`)
          : null}
      </div>
      <div className="m-tnm-sidebar__schedules">
        {headsign.times.map((time, idx) => {
          if (routeType === 2 && idx > 0) return null; // limit to 1 headsign
          return renderTime(time, headsign.name, routeType, idx);
        })}
      </div>
    </div>
  );
};

export default HeadsignComponent;
