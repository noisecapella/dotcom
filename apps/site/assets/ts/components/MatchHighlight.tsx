import React, { DetailedReactHTMLElement, useEffect, useState } from "react";

// React.createElement is used as opposed to the dangerouslySetInnerHTML prop to
// make the component contents easier to parse in testing

interface Props {
  text: string;
  matchQuery?: string;
}

// React component for highlighting matching text will wrap the first substring
// matching the matchQuery in a <b></b> and return the HTML result
const MatchHighlight = ({
  text,
  matchQuery
}: Props): DetailedReactHTMLElement<{}, HTMLElement> => {
  const [childNodes, setChildNodes] = useState<
    (DetailedReactHTMLElement<{}, HTMLElement> | string)[]
  >([text]);

  useEffect(() => {
    if (matchQuery && matchQuery !== "") {
      const matched = text.match(new RegExp(matchQuery, "i"));
      if (matched && matched[0]) {
        // split text on first instance of match only
        const [precedingText, ...moreTexts] = text.split(matched[0]);
        const subsequentText = moreTexts.join(matched[0]);
        setChildNodes([
          precedingText,
          React.createElement("b", { className: "u-highlight" }, matched[0]),
          subsequentText
        ]);
      }
    } else {
      setChildNodes([text]);
    }
  }, [text, matchQuery, setChildNodes]);

  return React.createElement("span", {}, ...childNodes);
};

export default MatchHighlight;
