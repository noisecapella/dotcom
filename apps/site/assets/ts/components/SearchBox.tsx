import React, { ReactElement, useState, useEffect } from "react";
import { searchIcon } from "../helpers/icon";

interface Props {
  id: string;
  labelText: string;
  onChange: (q: string) => void;
  placeholder?: string;
  className?: string;
}

const SearchBox = ({
  id,
  labelText,
  onChange,
  placeholder,
  className
}: Props): ReactElement<HTMLElement> | null => {
  const [query, setQuery] = useState("");

  useEffect(() => {
    onChange(query);
  }, [query, onChange]);

  return (
    <div className={`${className} c-form__input-container`}>
      <label htmlFor={id} className="sr-only">
        {labelText}
      </label>
      <input
        className="c-form__input"
        id={id}
        type="text"
        onChange={event => {
          setQuery(event.target.value);
        }}
        value={query}
        placeholder={placeholder || labelText}
      />
      {query !== "" && (
        <button
          className="c-form__reset-btn btn-link"
          aria-label="reset"
          type="button"
          onClick={() => {
            setQuery("");
          }}
        >
          <i
            aria-hidden="true"
            className="c-form__reset-icon fa fa-times-circle"
          />
        </button>
      )}
      <button
        className="c-form__submit-btn"
        aria-label="search"
        type="button"
        onClick={() => {
          setQuery(query);
        }}
      >
        {searchIcon("c-svg__icon-search-reverse-default")}
      </button>
    </div>
  );
};

export default SearchBox;
