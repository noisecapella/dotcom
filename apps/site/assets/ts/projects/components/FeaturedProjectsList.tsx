import React, { ReactElement } from "react";
import { Teaser as Project } from "../../__cms";
import FeaturedProjectsRow from "./FeaturedProjectsRow";

interface Props {
  projects: Project[];
  placeholderImageUrl: string;
}

const doGroupPairwise = (
  projectsToGroup: Project[],
  acc: Project[][]
): Project[][] => {
  if (projectsToGroup.length === 0) {
    return acc;
  }

  const nextPair = projectsToGroup.slice(0, 2);
  const rest = projectsToGroup.slice(2);
  return doGroupPairwise(rest, acc.concat([nextPair]));
};

const groupPairwise = (projectsToGroup: Project[]): Project[][] =>
  doGroupPairwise(projectsToGroup, []);

const FeaturedProjectsList = ({
  projects,
  placeholderImageUrl
}: Props): ReactElement<HTMLElement> | null => {
  if (projects.length === 0) {
    return null;
  }

  const groupedProjects = groupPairwise(projects);

  return (
    <div className="m-featured-project-list">
      <div className="page-section">
        {groupedProjects.map(projectPair => {
          const secondProjectId = projectPair[1] ? projectPair[1].id : "none";
          const key = `featured-projects-row-${projectPair[0].id}_${secondProjectId}`;
          return (
            <FeaturedProjectsRow
              key={key}
              placeholderImageUrl={placeholderImageUrl}
              projectPair={projectPair}
            />
          );
        })}
      </div>
    </div>
  );
};

export default FeaturedProjectsList;
