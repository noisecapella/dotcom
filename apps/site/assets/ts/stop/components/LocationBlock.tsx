import React, { ReactElement } from "react";
import RoutePillList from "./RoutePillList";
import { Stop } from "../../__v3api";
import { TypedRoutes } from "./__stop";
import renderSvg from "../../helpers/render-svg";

import streetViewSvg from "../../../static/images/icon-street-view-default.svg";

interface Props {
  routes: TypedRoutes[];
  stop: Stop;
  streetViewUrl: string | null;
}

const addressOrMunicipality = (stop: Stop): ReactElement | null => {
  if (stop.address)
    return (
      <div className="m-stop-page__location">
        <h3 className="u-small-caps">Address</h3>
        <div className="h3">{stop.address}</div>
      </div>
    );

  if (stop.municipality)
    return (
      <div className="m-stop-page__location">
        <h3 className="u-small-caps">City</h3>
        <div className="h3">{stop.municipality}</div>
      </div>
    );

  return null;
};

const latLngString = (stop: Stop): string =>
  `${stop.latitude},${stop.longitude}`;

const streetViewLink = (stop: Stop): string =>
  `https://www.google.com/maps/@?api=1&map_action=pano&viewpoint=${latLngString(
    stop
  )}`;

const LocationBlock = ({
  routes,
  stop,
  streetViewUrl
}: Props): ReactElement<HTMLElement> => (
  <div className="m-stop-page__location-block">
    {addressOrMunicipality(stop)}
    <div className="m-stop-page__location-links">
      <div className="m-stop-page__location-link">
        <a
          href={`/trip-planner/to/${(stop.id === "Boat-Hingham"
            ? stop.id
            : latLngString(stop)
          )
            .replace(/\s+/g, "-")
            .toLowerCase()}`}
          className="btn btn-primary"
          target="_blank"
          rel="noopener noreferrer"
        >
          Get directions to this station
        </a>
      </div>
      <div className="m-stop-page__location-link">
        <a
          href={streetViewUrl || streetViewLink(stop)}
          target="_blank"
          rel="noopener noreferrer"
        >
          {renderSvg(
            "c-svg__icon-street-view m-stop-page__street-view-icon",
            streetViewSvg
          )}
          Street View
        </a>
      </div>
    </div>
    <RoutePillList routes={routes} />
  </div>
);

export default LocationBlock;
