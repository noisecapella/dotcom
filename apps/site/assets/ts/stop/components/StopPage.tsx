import React, { ReactElement, useReducer } from "react";
import useInterval from "use-interval";
import StopMapContainer from "./StopMapContainer";
import { StopPageData, StopMapData, TypedRoutes } from "./__stop";
import { EnhancedRoute } from "../../__v3api";
import StopPageHeader from "./StopPageHeader";
import { reducer, initialState, Dispatch, updateRoutesAction } from "../state";
import Alerts from "../../components/Alerts";
import AlertsTab from "./AlertsTab";
import Sidebar from "./Sidebar";
import LocationBlock from "./LocationBlock";
import Departures from "./Departures";
import SuggestedTransfers from "./SuggestedTransfers";
import { isHighSeverityOrHighPriority } from "../../models/alert";

interface Props {
  stopPageData: StopPageData;
  mapData: StopMapData;
  mapId: string;
}

export const fetchData = (stopId: string, dispatch: Dispatch): Promise<void> =>
  window.fetch &&
  window
    .fetch(`/stops/api?id=${stopId}`)
    .then((response: Response) => {
      if (response.ok) return response.json();
      throw new Error(response.statusText);
    })
    .then((routes: TypedRoutes[]) => dispatch(updateRoutesAction(routes)))
    .catch(() => {});

export default ({
  stopPageData,
  mapData,
  mapId
}: Props): ReactElement<HTMLElement> => {
  const {
    stop,
    street_view_url: streetViewUrl,
    routes,
    // eslint-disable-next-line camelcase
    routes_with_direction: routesWithDirection,
    // eslint-disable-next-line camelcase
    retail_locations: retailLocations,
    // eslint-disable-next-line camelcase
    suggested_transfers: suggestedTransfers,
    alerts,
    alerts_tab: alertsTab,
    routes_and_alerts: routesAndAlerts,
    tab
  } = stopPageData;

  const [state, dispatch] = useReducer(reducer, initialState(tab, routes));

  useInterval(() => {
    // temporarily disabled while we debug CPU issues
    // fetchData(stop.id, dispatch);
  }, 15000);

  const highPriorityAlerts = alerts.filter(isHighSeverityOrHighPriority);

  return (
    <>
      <StopPageHeader
        stopPageData={stopPageData}
        dispatch={dispatch}
        selectedTab={state.selectedTab}
      />
      {state.selectedTab === "alerts" ? (
        <AlertsTab alertsTab={alertsTab} />
      ) : (
        <>
          <Alerts alerts={highPriorityAlerts} />
          <div className="container">
            <h2>Station Information</h2>
            <p>
              See upcoming departures, maps, and other features at this
              location.
            </p>
          </div>
          <div className="m-stop-page__hero">
            <StopMapContainer
              initialData={mapData}
              mapId={mapId}
              stop={stop}
              routes={routes.reduce(
                (
                  accumulator: EnhancedRoute[],
                  groupedRoutes: TypedRoutes
                ): EnhancedRoute[] =>
                  accumulator.concat(
                    groupedRoutes.routes.map(typedRoute => typedRoute.route)
                  ),
                []
              )}
              routesWithDirection={routesWithDirection}
            />
            <div className="m-stop-page__hero-photo" />
          </div>
          <div className="container">
            <div className="page-section">
              <div className="row">
                <div className="col-12 col-sm-10 col-sm-offset-1 col-lg-7 col-lg-offset-0">
                  <LocationBlock
                    stop={stop}
                    routes={routes}
                    streetViewUrl={streetViewUrl}
                  />
                  <Departures
                    routes={state.routes}
                    routesAndAlerts={routesAndAlerts}
                    stop={stop}
                    selectedModes={state.selectedModes}
                    dispatch={dispatch}
                  />
                  <SuggestedTransfers suggestedTransfers={suggestedTransfers} />
                </div>
                <div className="col-12 col-sm-10 col-sm-offset-1 col-lg-4 col-lg-offset-1">
                  <Sidebar
                    dispatch={dispatch}
                    expandedBlocks={state.expandedBlocks}
                    focusedBlock={state.focusedBlock}
                    stop={stop}
                    routes={routes}
                    retailLocations={retailLocations}
                  />
                </div>
              </div>
            </div>
          </div>
        </>
      )}
    </>
  );
};
