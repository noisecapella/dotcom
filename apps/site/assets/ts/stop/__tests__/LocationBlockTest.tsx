import React from "react";
import renderer from "react-test-renderer";
import { shallow } from "enzyme";
import stopData from "./stopData.json";
import { StopPageData } from "../components/__stop";
import { Stop } from "../../__v3api";
import LocationBlock from "../components/LocationBlock";

const data = JSON.parse(JSON.stringify(stopData)) as StopPageData;

it("renders", () => {
  document.body.innerHTML =
    '<div><div id="react-root"><div id="test"></div></div></div>';

  const tree = renderer
    .create(
      <LocationBlock
        routes={data.routes}
        stop={data.stop}
        streetViewUrl={null}
      />
    )
    .toJSON();

  expect(tree).toMatchSnapshot();
});

it("uses lat/lng when linking to the trip planner", () => {
  expect(data.stop.latitude).toEqual(42.352271);
  expect(data.stop.longitude).toEqual(-71.055242);
  const latLng = `${data.stop.latitude},${data.stop.longitude}`;
  expect(
    shallow(<LocationBlock routes={[]} stop={data.stop} streetViewUrl={null} />)
      .find(".btn.btn-primary")
      .prop("href")
  ).toEqual(`/trip-planner/to/${latLng}`);
});

it("falls back to municipality if stop has no address", () => {
  const stopWithoutAddress = { ...data.stop, address: null };

  const wrapper = shallow(
    <LocationBlock routes={[]} stop={stopWithoutAddress} streetViewUrl={null} />
  );

  expect(wrapper.find(".m-stop-page__location .h3").text()).toEqual("Boston");
});

it("does not render location if stop has no address or municipality", () => {
  const stopWithoutLocation = {
    ...data.stop,
    address: null,
    municipality: null
  };

  const wrapper = shallow(
    <LocationBlock
      routes={[]}
      stop={stopWithoutLocation}
      streetViewUrl={null}
    />
  );

  expect(wrapper.find(".m-stop-page__location").exists()).toEqual(false);
  expect(wrapper.find(".m-stop-page__location-links").exists()).toEqual(true);
});
