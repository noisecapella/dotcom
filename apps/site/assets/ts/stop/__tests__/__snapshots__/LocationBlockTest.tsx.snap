// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders 1`] = `
<div
  className="m-stop-page__location-block"
>
  <div
    className="m-stop-page__location"
  >
    <h3
      className="u-small-caps"
    >
      Address
    </h3>
    <div
      className="h3"
    >
      700 Atlantic Ave, Boston, MA 02110
    </div>
  </div>
  <div
    className="m-stop-page__location-links"
  >
    <div
      className="m-stop-page__location-link"
    >
      <a
        className="btn btn-primary"
        href="/trip-planner/to/42.352271,-71.055242"
        rel="noopener noreferrer"
        target="_blank"
      >
        Get directions to this station
      </a>
    </div>
    <div
      className="m-stop-page__location-link"
    >
      <a
        href="https://www.google.com/maps/@?api=1&map_action=pano&viewpoint=42.352271,-71.055242"
        rel="noopener noreferrer"
        target="_blank"
      >
        <span
          aria-hidden="true"
          className="notranslate c-svg__icon-street-view m-stop-page__street-view-icon"
          dangerouslySetInnerHTML={
            Object {
              "__html": "SVG",
            }
          }
        />
        Street View
      </a>
    </div>
  </div>
  <div
    className="m-route-pills"
  >
    <div
      className="m-route-pills--subway"
    >
      <a
        className="
      m-stop-page__header-feature
      m-stop-page__header-description
      u-small-caps
      u-bg--red-line
    "
        href="/schedules/Red"
      >
        Red Line
      </a>
    </div>
    <div
      className="m-route-pills--bus"
    >
      <a
        className="
      m-stop-page__header-feature
      m-stop-page__header-description
      u-small-caps
      u-bg--silver-line
    "
        href="/schedules/741"
      >
        SL1
      </a>
      <a
        className="
      m-stop-page__header-feature
      m-stop-page__header-description
      u-small-caps
      u-bg--silver-line
    "
        href="/schedules/742"
      >
        SL2
      </a>
      <a
        className="
      m-stop-page__header-feature
      m-stop-page__header-description
      u-small-caps
      u-bg--silver-line
    "
        href="/schedules/743"
      >
        SL3
      </a>
    </div>
    <div
      className="m-route-pills--commuter_rail"
    >
      <a
        className="
      m-stop-page__header-feature
      m-stop-page__header-description
      u-small-caps
      u-bg--commuter-rail
    "
        href="/schedules/CR-Fairmount"
      >
        Fairmount Line
      </a>
      <a
        className="
      m-stop-page__header-feature
      m-stop-page__header-description
      u-small-caps
      u-bg--commuter-rail
    "
        href="/schedules/CR-Worcester"
      >
        Framingham/​Worcester Line
      </a>
      <a
        className="
      m-stop-page__header-feature
      m-stop-page__header-description
      u-small-caps
      u-bg--commuter-rail
    "
        href="/schedules/CR-Franklin"
      >
        Franklin Line
      </a>
      <a
        className="
      m-stop-page__header-feature
      m-stop-page__header-description
      u-small-caps
      u-bg--commuter-rail
    "
        href="/schedules/CR-Greenbush"
      >
        Greenbush Line
      </a>
      <a
        className="
      m-stop-page__header-feature
      m-stop-page__header-description
      u-small-caps
      u-bg--commuter-rail
    "
        href="/schedules/CR-Kingston"
      >
        Kingston/​Plymouth Line
      </a>
      <a
        className="
      m-stop-page__header-feature
      m-stop-page__header-description
      u-small-caps
      u-bg--commuter-rail
    "
        href="/schedules/CR-Middleborough"
      >
        Middleborough/​Lakeville Line
      </a>
      <a
        className="
      m-stop-page__header-feature
      m-stop-page__header-description
      u-small-caps
      u-bg--commuter-rail
    "
        href="/schedules/CR-Needham"
      >
        Needham Line
      </a>
      <a
        className="
      m-stop-page__header-feature
      m-stop-page__header-description
      u-small-caps
      u-bg--commuter-rail
    "
        href="/schedules/CR-Providence"
      >
        Providence/​Stoughton Line
      </a>
    </div>
  </div>
</div>
`;
